src/main/kotlin/app/cash/licensee/task.kt

```diff
@@ -18,56 +18,92 @@ package app.cash.licensee
 import app.cash.licensee.ViolationAction.FAIL
 import app.cash.licensee.ViolationAction.IGNORE
 import java.io.File
-import kotlin.properties.Delegates
 import kotlinx.serialization.builtins.ListSerializer
 import kotlinx.serialization.json.Json
 import org.gradle.api.DefaultTask
 import org.gradle.api.artifacts.Configuration
-import org.gradle.api.attributes.Usage
-import org.gradle.api.attributes.Usage.USAGE_ATTRIBUTE
-import org.gradle.api.file.FileCollection
+import org.gradle.api.artifacts.ResolvedArtifact
+import org.gradle.api.artifacts.component.ModuleComponentIdentifier
+import org.gradle.api.attributes.Attribute
+import org.gradle.api.file.DirectoryProperty
 import org.gradle.api.logging.LogLevel.ERROR
 import org.gradle.api.logging.LogLevel.INFO
 import org.gradle.api.logging.LogLevel.LIFECYCLE
 import org.gradle.api.logging.LogLevel.WARN
 import org.gradle.api.logging.Logger
 import org.gradle.api.logging.Logging
-import org.gradle.api.tasks.Classpath
+import org.gradle.api.provider.MapProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.OutputDirectory
 import org.gradle.api.tasks.TaskAction
 
 internal open class LicenseeTask : DefaultTask() {
   @get:Input
-  lateinit var dependencyConfig: DependencyConfig
+  val dependencyConfig: Property<DependencyConfig> = project.objects.property(DependencyConfig::class.java)
```

Comment:
I can change this myself separately, but nowadays I prefer `abstract class` + `abstract val`. The implicit subclass and the weird availability of `project` to create these instances despite not being attached to a project is too much magic for my taste. With everything `abstract` at least you are delegating all behavior to the runtime-generated subclass and it's 100% out-of-sight!

Reply:
Nice, I do like it more, too. But was too lazy to change it.